File: rb/features.rb
# OgliLinkShortener SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module OgliLinkShortenerFeatures
  def self.make_feature(name)
    case name
    when "base"
      OgliLinkShortenerBaseFeature.new
    when "test"
      OgliLinkShortenerTestFeature.new
    else
      OgliLinkShortenerBaseFeature.new
    end
  end
end

File: py/features.py
# OgliLinkShortener SDK feature factory

from feature.base_feature import OgliLinkShortenerBaseFeature
from feature.test_feature import OgliLinkShortenerTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OgliLinkShortenerBaseFeature(),
        "test": lambda: OgliLinkShortenerTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
